internal/app/machined/pkg/controllers/config/acquire_test.go

```diff
@@ -0,0 +1,455 @@
+			ConfigSetter:          s.configSetter,
+			EventPublisher:        s.eventPublisher,
+			ValidationMode:        validationModeMock{},
+			ConfigPath:            s.configPath,
+		}))
+	}
+
+	suite.Run(t, s)
+}
+
+func (suite *AcquireSuite) triggerAcquire() {
+	suite.Require().NoError(suite.State().Create(suite.Ctx(), v1alpha1.NewAcquireConfigSpec()))
+}
+
+func (suite *AcquireSuite) waitForConfig() config.Provider {
+	cfg, ok := channel.RecvWithContext(suite.Ctx(), suite.configSetter.cfgCh)
+	suite.Require().True(ok)
+
+	status := v1alpha1.NewAcquireConfigStatus()
+	rtestutils.AssertResources(suite.Ctx(), suite.T(), suite.State(), []resource.ID{status.Metadata().ID()}, func(*v1alpha1.AcquireConfigStatus, *assert.Assertions) {})
+
+	return cfg
+}
+
+func (suite *AcquireSuite) injectViaMaintenance(cfg []byte) {
+	_, err := suite.State().WatchFor(suite.Ctx(), runtime.NewMaintenanceServiceRequest().Metadata(), state.WithEventTypes(state.Created))
+	suite.Require().NoError(err)
+
+	mCfg, err := configloader.NewFromBytes(cfg)
+	suite.Require().NoError(err)
+
+	suite.Require().NoError(suite.State().Create(suite.Ctx(), configresource.NewMachineConfigWithID(mCfg, configresource.MaintenanceID)))
+
+	_, err = suite.State().WatchFor(suite.Ctx(), runtime.NewMaintenanceServiceRequest().Metadata(), state.WithEventTypes(state.Destroyed))
+	suite.Require().NoError(err)
+}
+
+func (suite *AcquireSuite) TestFromDisk() {
+	suite.Require().NoError(os.WriteFile(suite.configPath, suite.completeMachineConfig, 0o644))
+
+	suite.triggerAcquire()
+
+	cfg := suite.waitForConfig()
+	suite.Require().Equal(cfg.Cluster().Name(), suite.clusterName)
+
+	suite.Assert().Empty(suite.eventPublisher.getEvents())
+	suite.Assert().Equal(
+		[]platform.Event{
+			{
+				Type:    platform.EventTypeConfigLoaded,
+				Message: "Talos machine config loaded successfully.",
+			},
+		},
+		suite.platformEvent.getEvents(),
+	)
+}
+
+func (suite *AcquireSuite) TestFromDiskFailure() {
+	suite.Require().NoError(os.WriteFile(suite.configPath, append([]byte("aaa"), suite.completeMachineConfig...), 0o644))
+
+	suite.triggerAcquire()
+
+	suite.AssertWithin(time.Second, 10*time.Millisecond, func() error {
+		if len(suite.platformEvent.getEvents()) == 0 || len(suite.eventPublisher.getEvents()) == 0 {
+			return retry.ExpectedErrorf("no events received")
+		}
+
+		return nil
+	})
+
+	ev := suite.platformEvent.getEvents()[0]
+	suite.Assert().Equal(platform.EventTypeFailure, ev.Type)
+	suite.Assert().Equal("Error loading and validating Talos machine config.", ev.Message)
+	suite.Assert().Equal("failed to load config from STATE: unknown keys found during decoding:\naaaversion: v1alpha1 # Indicates the schema used to decode the contents.\n", ev.Error.Error())
+
+	suite.Assert().Equal(&machineapi.ConfigLoadErrorEvent{
+		Error: "failed to load config from STATE: unknown keys found during decoding:\naaaversion: v1alpha1 # Indicates the schema used to decode the contents.\n",
```

Comment:
error seems weird, we only wrote `aaa`, why does error says v1alpha1 :thinking:

Reply:
we prepend 'aaa' to the machine config, so that it was:

```yaml
version: v1alpha1
...
```

and becomes:

```yaml
aaaversion: v1alpha1
...
```

Reply:
ahh I completely missed the prepending, now I see it